File: 53-maximum-subarray/maximum-subarray.c
int maxSubArray(int* nums, int numsSize) {
    int sum = 0,prevsum= -1000000;
    for(int i=0; i<numsSize; i++){
        int tot = sum + nums[i];
        if(nums[i] > tot) sum = nums[i];
        else sum = tot;
        if(prevsum < sum) prevsum = sum ;

    }
    return prevsum; 
}

File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        msf = nums[0]
        mtl = nums[0]
        for i in nums[1:]:
            mtl = max(i, mtl+i)
            msf= max(mtl, msf)
        return msf            